sensor/kubernetes/certrefresh/certificates/requester.go

```diff
@@ -140,44 +110,24 @@ func (f *localScannerMessageFactory) newMsgFromSensor(requestID string) *central
 	}
 }
 
-// Start makes the certificate requester listen to `receiveC` and forwards responses to any request that is running
-// as a call to RequestCertificates.
-func (r *genericRequester[ReqT, RespT]) Start() {
-	r.stopC.Reset()
-	go r.dispatchResponses()
-}
-
-// Stop makes the certificate stop forwarding responses to running requests. Subsequent calls to RequestCertificates
-// will fail with ErrCertificateRequesterStopped.
-// Currently active calls to RequestCertificates will continue running until cancelled or timed out via the
-// provided context.
-func (r *genericRequester[ReqT, RespT]) Stop() {
-	r.stopC.Signal()
-}
-
-func (r *genericRequester[ReqT, RespT]) dispatchResponses() {
-	for {
-		select {
-		case <-r.stopC.Done():
-			return
-		case msg := <-r.receiveC:
-			responseC, ok := r.requests.Load(msg.GetRequestId())
-			if !ok {
-				log.Debugf("request ID %q does not match any known request ID, dropping response",
-					msg.GetRequestId())
-				continue
-			}
-			r.requests.Delete(msg.GetRequestId())
-			// Doesn't block even if the corresponding call to RequestCertificates is cancelled and no one
-			// ever reads this, because requestC has buffer of 1, and we removed it from `r.requests` above,
-			// in case we get more than 1 response for `msg.GetRequestId()`.
-			responseC.(chan RespT) <- msg
-		}
+// DispatchResponse forwards a response from Central to a RequestCertificates call running in another goroutine
+func (r *genericRequester[ReqT, RespT]) DispatchResponse(response *Response) {
+	if !r.ongoingRequest.Load() {
+		log.Debugf("Received a response for request ID %q, but there is no ongoing RequestCertificates call.", response.RequestId)
+		return
 	}
+	if r.ongoingRequestID != response.RequestId {
+		log.Debugf("Request ID %q does not match ongoing request ID %q.",
+			response.RequestId, r.ongoingRequestID)
+		return
```

Comment:
Shall we make those two log lines "louder"? I would suggest at least a `log.Warnf`, but maybe we shall even go for `util.Must`, so that we panic in the dev builds.